.gitignore

```diff
@@ -21,3 +21,4 @@ attack-theme/templates/tactics
 attack-theme/templates/techniques
 attack-theme/static/scripts/settings.js
 content/pages/
+data/stix_replacement.js
```

Comment:
Shouldn't `stix_replacement.js` be `stix_replacement.json`? I don't think it should be a `js` file unless it's either A) imported in the UI to quickly load data (which can also be done with json, FWIW), or B) contains executable code. This file is as far as I can tell neither, and should therefore be a json file.

Reply:
My understanding is that stix_replacement is a way to move data from the python args into pelican. If so, we should consider making a more permanent/extensible way to do this since we'll likely have to do this again in the future.

For example:
1. have a function called "send_to_pelican" which takes as argument key:value pairs. This can be called by any module to queue config settings to send to the pelican process for use in filters or whatnot.
2. In `website_build.py`, the execution of `pelican_content` can then retrieve those settings, write them to a temporary JSON file (e.g `data/pelican_settings.json`). 
3. `pelicanconf.py` can then read that file once, before getting to any of the filter functions, to get the full list of settings that was sent to it from the modules
4. Once pelican exits and we return to pelican_content(), before returning we can delete the temporary `data/pelican_settings.json` file since it's no longer needed.

Reply:
Changes staged for review.